sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -270,52 +269,17 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         ifNotExists = false)).toRdd
   }
 
-  private def normalizedParCols: Option[Seq[String]] = partitioningColumns.map { cols =>
-    cols.map(normalize(_, "Partition"))
-  }
-
-  private def normalizedBucketColNames: Option[Seq[String]] = bucketColumnNames.map { cols =>
-    cols.map(normalize(_, "Bucketing"))
-  }
-
-  private def normalizedSortColNames: Option[Seq[String]] = sortColumnNames.map { cols =>
-    cols.map(normalize(_, "Sorting"))
-  }
-
   private def getBucketSpec: Option[BucketSpec] = {
     if (sortColumnNames.isDefined) {
       require(numBuckets.isDefined, "sortBy must be used together with bucketBy")
     }
 
-    for {
-      n <- numBuckets
-    } yield {
+    numBuckets.map { n =>
       require(n > 0 && n < 100000, "Bucket number must be greater than 0 and less than 100000.")
```

Comment:
Orthogonal to your PR: This means Spark supports buckets in range [1, 99999]. Any reason to have a low value for upper bound ?

Also, I don't think this code gets executed if the bucketed table is written via SQL. The only check I can see was when we create `BucketSpec` but its for lower bound only : https://github.com/apache/spark/blob/4d4d0de7f64cefbca28dc532b7864de9626aa241/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala#L138 . This check should be only present in `BucketSpec` creation to be consistent across the codebase.